From 014230828b46536ab7f74897407b9de3f4235ed2 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Sun, 20 Mar 2016 15:33:50 +0200
Subject: [PATCH 0156/1345] arm64: dts: marvell: use new clock binding on
 Armada AP806

commit  d6ffeb490e8494be1e26c80d89f546e7950668ed from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Moved clock node of AP-806-Z to DTSI of Z1 chip,
  and update the nodes of the Z1 units to use Z1 clock driver

Change-Id: Ie8750a1f5358770131318e0ebba8919c8b6e05ae
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28363
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/marvell/armada-ap806-z1-quad.dtsi     |   36 +++++++++++++++++++-
 1 file changed, 35 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi
index 49e901f..4a10541 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi
@@ -81,7 +81,41 @@
 	};
 
 	ap806 {
-		config-space {
+		config-space@f0000000 {
+			/* For SPI, I2C, and Serial units override the clock */
+			spi0: spi@510600 {
+				clocks = <&ringclk 2>;
+			};
+
+			i2c0: i2c@511000 {
+				clocks = <&ringclk 2>;
+			};
+
+			serial@512000 {
+				clocks = <&ringclk 2>;
+			};
+
+			serial@512100 {
+				clocks = <&ringclk 2>;
+
+			};
+			/* Add clock node for AP-806-Z SoC
+			** The Z1 revision use different driver than A0 revision
+			*/
+			coreclk: clk@0x6F8204 {
+				compatible = "marvell,armada-ap806-core-clock";
+				reg = <0x6F8204 0x04>;
+				#clock-cells = <1>;
+				clock-output-names = "ddr", "ring", "cpu";
+			};
+
+			ringclk: clk@0x6F8250 {
+				compatible = "marvell,armada-ap806-ring-clock";
+				reg = <0x6F8250 0x04>;
+				#clock-cells = <1>;
+				clock-output-names = "ring-0", "ring-2", "ring-3", "ring-4", "ring-5";
+				clocks = <&coreclk 1>;
+			};
 			xor0@400000 {
 				compatible = "marvell,mv-xor-v2-z1";
 			};
-- 
1.7.9.5

